src/lib/Scenes/Fair/Screens/FairBMWArtActivation.tsx

```diff
@@ -0,0 +1,147 @@
+import styled from "styled-components/native"
+import { defaultEnvironment } from "../../../relay/createEnvironment"
+import renderWithLoadProgress from "../../../utils/renderWithLoadProgress"
+
+interface Props extends ViewProperties {
+  fair: FairBMWArtActivationQuery["response"]["fair"]
+}
+
+interface State {
+  sections: Array<{
+    type: "art-activation" | "press-release"
+    data: any
+  }>
+}
+
+interface ShowMoreMetadataForFairs {
+  sponsoredContent?: { pressReleaseUrl?: string; activationText?: string }
+}
+
+export const shouldShowFairBMWArtActivationLink = (data: ShowMoreMetadataForFairs) => {
+  return data.sponsoredContent
+}
+
+export class FairBMWArtActivation extends React.Component<Props, State> {
+  state = {
+    sections: [],
+  }
+
+  componentDidMount() {
+    const {
+      fair: { sponsoredContent },
+    } = this.props
+    const sections = []
+
+    if (sponsoredContent.activationText) {
+      sections.push({
+        type: "art-activation",
+        data: {
+          activationText: sponsoredContent.activationText,
+        },
+      })
+    }
+
+    if (sponsoredContent.pressReleaseUrl) {
+      sections.push({
+        type: "press-release",
+        data: {
+          pressReleaseUrl: sponsoredContent.pressReleaseUrl,
+        },
+      })
+    }
+
+    this.setState({ sections })
+  }
+
+  renderItemSeparator = () => <Box py={3} px={2} />
+
+  handleViewPressRelease(url) {
+    SwitchBoard.presentNavigationViewController(this, url)
+  }
+  renderItem = ({ item: { data, type } }) => {
+    switch (type) {
+      case "art-activation":
+        return (
+          <>
+            <Serif size="3" lineHeight="20">
+              {data.activationText}
+            </Serif>
+          </>
+        )
+      case "press-release":
+        return (
+          <>
+            {data.pressReleaseUrl && (
+              <CaretButton
+                text="View press release"
+                onPress={() => this.handleViewPressRelease(data.pressReleaseUrl)}
+              />
+            )}
+          </>
+        )
+    }
+  }
+
+  render() {
+    return (
+      <Theme>
+        <FlatList
+          data={this.state.sections}
+          renderItem={item => <Box px={2}>{this.renderItem(item)}</Box>}
+          ListHeaderComponent={
+            <>
+              <PressReleaseContainer>
+                <Box>
+                  <Logo source={require("../../../../../Pod/Assets/assets/images/BMW-logo.jpg")} />
+                </Box>
+                <Sans size="3" px={1} weight="medium">
+                  BMW Art Activations
+                </Sans>
+              </PressReleaseContainer>
+            </>
+          }
+          ItemSeparatorComponent={this.renderItemSeparator}
+          keyExtractor={(item, index) => item.type + String(index)}
+        />
+      </Theme>
+    )
+  }
+}
+
+export const FairBMWArtActivationRenderer: React.SFC<{ fairID: string }> = ({ fairID }) => (
```

Comment:
Usually, we put the query renderer component into the `QueryRenderers.tsx` file or somewhere separate from the component, and then use `createFragmentContainer` to wrap the plain React component (in this case, `FairBMWArtActivation`). This lets us separate out the query (`query FairBMWArtActivationQuery($fairID: String!) { ...`) from the data fragment.

Is there a reason we'd like to do things this way instead? I'm happy to pair as well 😄

Reply:
We should probably discuss this in our platform meeting on async in slack or a RFC as it's proposed change in how we organize our code. But `QueryRenderers.tsx` is getting fairly large (> 500 LOC) so we've been co locating QueryRenderers with their corresponding top level relay fragment / react components for our Local Discovery Component VCs.

I just synced with @ashleyjelks as she was having issues after moving the QueryRenderer, I suggested we leave it there and sync later on best practices around code organization.

Reply:
Sure, if this works for you.